odbc/src/api/functions.rs

```diff
@@ -1020,6 +1020,22 @@ pub unsafe extern "C" fn SQLDisconnect(connection_handle: HDbc) -> SqlReturn {
         || {
             let conn_handle = MongoHandleRef::from(connection_handle);
             let conn = must_be_valid!((*conn_handle).as_connection());
+
+            // close open cursors on any statements that haven't been closed
+            if let Ok(stmts) = conn.statements.write() {
```

Comment:
We need to drop the statements too based on the [transition table for SQLDisconnect]( https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/statement-transitions?view=sql-server-ver16#sqldisconnect).
It more or less does a SQLFreeHandle(stmt) for each stmt behind the scene.
And I think we should close the cursor for the stmt in SQLFreeHandle as well, no need to keep it open.


Note
SQLFreeStmt doesn't drop the statements only closes them. They can then be reused with another call to SQLExecute for example.